add sig cloud provider leads to owners

elmiko opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issue-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/guide/release-notes.md
6. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**
/kind cleanup


**What this PR does / why we need it**:

as discussed in the 18 June 2025 office hours meeting for sig cloud provider, we agreed to start adding the sig leads to the owners files. this is being done to help with future maintenance.


**Which issue(s) this PR fixes**:
<!-- 
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
<!--  
If no, just write "NONE".
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
2. 
-->
```release-note
NONE
```

olemarkus commented:
Can we hold of changing my status for a few weeks? I still keep tabs on this project and I'd like to contribute if I can. I'll report here if that is possible or not.

elmiko commented:
thanks @kmala , i'm happy to queue up the emeritus changes as well.

@olemarkus no rush on our part, we can certainly keep this PR open for a few weeks if that helps.

elmiko commented:
thanks @olemarkus !

i've moved @nckturner , @hakman , and @kishorj to the emeritus approvers, and removed them from the approvers and reviewers. i have left @olemarkus in place, and have added the sig leads to the approvers and reviewers.
